tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/AfterReceiveRequestIntegration.cs

```diff
@@ -0,0 +1,98 @@
+// <copyright file="AfterReceiveRequestIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+#if NETFRAMEWORK
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Wcf
+{
+    /// <summary>
+    /// System.ServiceModel.Dispatcher.ImmutableDispatchRuntime.AfterReceiveRequest calltarget instrumentation.
+    /// This integration starts the WCF server span when DD_TRACE_DELAY_WCF_INSTRUMENTATION_ENABLED=true.
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "System.ServiceModel",
+        TypeName = "System.ServiceModel.Dispatcher.ImmutableDispatchRuntime",
+        MethodName = "AfterReceiveRequest",
+        ReturnTypeName = ClrNames.Void,
+        ParameterTypeNames = new[] { "System.ServiceModel.Dispatcher.MessageRpc&" },
+        MinimumVersion = "4.0.0",
+        MaximumVersion = "4.*.*",
+        IntegrationName = WcfCommon.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class AfterReceiveRequestIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TMessageRpc">Type of the rpc message</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="rpc">MessageRpc instance</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TMessageRpc>(TTarget instance, ref TMessageRpc rpc)
+        {
+            if (!Tracer.Instance.Settings.IsIntegrationEnabled(WcfCommon.IntegrationId) || !Tracer.Instance.Settings.DelayWcfInstrumentationEnabled || WcfCommon.GetCurrentOperationContext is null)
```

Comment:
nit: I'd extract all the `Tracer.Instance` to a variable; technically it can change out from under you 😄

Reply:
Good call, done in 93d6867e10bb95520f520905deb8bd88c85abbc8